types/src/system/auction/bid.rs

```diff
@@ -237,15 +264,20 @@ impl Bid {
 
         let mut initialized = false;
 
-        if vesting_schedule.initialize(staked_amount) {
+        if vesting_schedule
+            .initialize_with_weekly_schedule(staked_amount, vesting_schedule_period_millis)
```

Comment:
i don't think you should be renaming this to weekly_schedule; the mechanism should not make assumptions about the period